probe-rs/src/debug/language/c.rs

```diff
@@ -137,6 +180,64 @@ fn print_arbitrary_length(is_signed: bool, num: &mut [u8]) -> String {
     out
 }
 
+// Divide byte-by-byte
+fn divide(num: &mut [u8], by: u8) -> u32 {
+    let by = by as u32;
+
+    let mut carry = 0;
+    for byte in num.iter_mut().rev() {
+        let val = *byte as u32 + carry * 256;
+        *byte = (val / by) as u8;
+        carry = val % by;
+    }
+
+    carry
+}
+
+// Multiply byte-by-byte
+fn multiply(num: &mut Vec<u8>, by: u8) {
+    let by = by as u32;
+
+    let mut carry = 0;
+    for byte in num.iter_mut() {
+        let val = *byte as u32 * by + carry;
+        *byte = (val & 0xFF) as u8;
+        carry = val >> 8;
+    }
+
+    if carry > 0 {
+        num.push(carry as u8);
+    }
+}
+
+/// A very naive implementation of parsing an arbitrary length number.
+fn parse_arbitrary_length(input: &str) -> Result<Vec<u8>, DebugError> {
```

Comment:
Is there an issue with https://crates.io/crates/parse_int? :)

Reply:
Not really, I just didn't want to limit us to `u128`. I guess that may be a premature worry anyway, and oh crap I'm not handling the case where the user input is wider than the variable.